src/abstractdataframe/show.jl

```diff
@@ -675,46 +681,42 @@ function _show(io::IO,
 
                 v = col[kr]
 
-                order_v = 0
+                v_str = sprint(print, v, context = :compact => compact_printing)
 
-                if v !== missing
-                    abs_v = abs(v)
-                    log_v = !isinf(v) && !isnan(v) && abs_v > 1 ?
-                        floor(Int, log10(abs_v))::Int : 0
+                # We want to align everything at '.'.
+                id_dp = findfirst('.', v_str)
 
-                    # If the order is higher than 5, then we print using
-                    # scientific notation.
-                    order_v = log_v > 5 ? 0 : floor(Int, log_v)
-
-                    # If the number is negative (including -0.0), we need to add an additional
-                    # padding to print the sign.
-                    isless(v, 0) && (order_v += 1)
+                if id_dp !== nothing
+                    (align_col_i < id_dp) && (align_col_i = id_dp)
+                else
+                    id_dp = 0
```

Comment:
I understand this should not happen in theory - right?

Reply:
Yes, it can happen if the value is `missing` for example.